k8s-scheduler/src/test/java/com/vmware/dcm/ScopeTest.java

```diff
@@ -0,0 +1,280 @@
+        for (int i = 0; i < numNodes; i++) {
+            final String nodeName = "n" + i;
+            final Map<String, String> nodeLabels = new HashMap<>();
+            if (gpuNodesIdx.contains(i)) {
+                nodeLabels.put("gpu", "true");
+            }
+            if (ssdNodesIdx.contains(i)) {
+                nodeLabels.put("ssd", "true");
+            }
+            final Node node = newNode(nodeName, nodeLabels, Collections.emptyList());
+
+            // nodes have more spare capacity as i increases (favoring unlabeled nodes)
+            node.getStatus().getCapacity().put("cpu", new Quantity(String.valueOf(10 + 10 * i / numNodes)));
+            node.getStatus().getCapacity().put("memory", new Quantity(String.valueOf(1000 + (1000 * i) / numNodes)));
+            node.getStatus().getCapacity().put("pods", new Quantity(String.valueOf(100)));
+            nodeResourceEventHandler.onAddSync(node);
+        }
+
+        final List<Integer> gpuPodsIdx = List.of(0, 1, 2, 3, 4, 5, 6, 7);
+        final List<Integer> ssdPodsIdx = List.of(2, 3, 4, 5, 6, 7, 8, 9);
+        for (int i = 0; i < numPods; i++) {
+            final String podName = "p" + i;
+            final Map<String, String> selectorLabels = new HashMap<>();
+            if (gpuPodsIdx.contains(i)) {
+                selectorLabels.put("gpu", "true");
+            }
+            if (ssdPodsIdx.contains(i)) {
+                selectorLabels.put("ssd", "true");
+            }
+            final Pod newPod = newPod(podName, UUID.randomUUID(), "Pending", selectorLabels, Collections.emptyMap());
+            handler.onAddSync(newPod);
+        }
+
+        final Scheduler scheduler = new Scheduler(dbConnectionPool, policies, "ORTOOLS", true, NUM_THREADS);
+        scheduler.setScopeOn();
+        scheduler.scheduleAllPendingPods(new EmulatedPodToNodeBinder(dbConnectionPool));
+
+        // Check that all pods have been scheduled to a node eligible by the scope filtering
+        final Result<PodInfoRecord> fetch = conn.selectFrom(Tables.POD_INFO).fetch();
+        assertEquals(numPods, fetch.size());
+        fetch.forEach(e -> assertTrue(e.getNodeName() != null
+                // gpuPod => gpuNode
+                && (!gpuPodsIdx.contains(Integer.parseInt(e.getPodName().substring(1)))
+                || gpuNodesIdx.contains(Integer.parseInt(e.getNodeName().substring(1))))
+                // ssdPod => ssdNode
+                && (!ssdPodsIdx.contains(Integer.parseInt(e.getPodName().substring(1)))
+                || ssdNodesIdx.contains(Integer.parseInt(e.getNodeName().substring(1))))));
+    }
+
+
+    private Model buildModel(final DSLContext conn) {
+        final OrToolsSolver orToolsSolver = new OrToolsSolver.Builder()
+                .setPrintDiagnostics(true)
+                .build();
+        return Model.build(conn, orToolsSolver, Policies.getDefaultPolicies());
+    }
+
+    static Pod newPod(final String name) {
```

Comment:
Aren't these newPod()/newNode() utilities the same as from SchedulerTest? If so, just call SchedulerTest.newPod/newNode.